bootstrap/src/main/java/io/airlift/bootstrap/ConfigurationException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.bootstrap;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.internal.Messages;
+import com.google.inject.spi.Message;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+public final class ConfigurationException
+        extends RuntimeException
+{
+    private final Set<Message> errors;
+    private final Set<Message> warnings;
+    private final List<Message> allMessages;
+
+    public ConfigurationException(Collection<Message> errors, Collection<Message> warnings)
+    {
+        this.errors = ImmutableSet.copyOf(errors);
+        this.warnings = ImmutableSet.copyOf(warnings);
+
+        ImmutableList.Builder<Message> allMessages = ImmutableList.builder();
+        errors.stream()
+                .map(error -> new Message(error.getSources(), "Error: " + error.getMessage(), error.getCause()))
+                .forEach(allMessages::add);
+        warnings.stream()
+                .map(warning -> new Message(warning.getSources(), "Warning: " + warning.getMessage(), warning.getCause()))
```

Comment:
Since this flies as an exception, a "warning" may be a misunderstood. (When i see a warning, i usually expect the problem will try to continue).

Maybe instead "error, warning" naming use "fatal, error" for discerning them?

Reply:
This will show errors first, then warnings. We could add a validation to show this always has errors.
```java
checkArgument(!errors.isEmpty(), "no errors present");